chain/chain/src/types.rs

```diff
@@ -491,7 +491,7 @@ pub trait RuntimeAdapter: Send + Sync {
         shard_id: ShardId,
         state_root: &StateRoot,
         part_id: PartId,
-        data: &[u8],
+        part: &StatePart,
```

Comment:
Given this is a validate function, I'm wondering whether it makes sense to change the signature to take in a byte array of data and return an optional StatePart? That way we can encapsulate the borsh deserialization and potential decompression into the validation step?

cc @wacban

Reply:
Added TODO comment and created the follow-up issue: https://github.com/near/nearcore/issues/14124